src/cmd/init.rs

```diff
@@ -76,23 +99,107 @@ pub fn run_init(
+                    let sw_file = cmd::get_seed_words_file(wallet_dir);
+                    fs::write(sw_file.clone(), seed_words.clone())?;
+                    WalletKey::SeedWordsFile {}
+                }
+                None => {
+                    let n_words = MnemonicType::for_word_count(n_words)?;
+                    let seed_words: GeneratedKey<_, Segwitv0> =
+                        Mnemonic::generate((n_words, Language::English))
+                            .map_err(|_| anyhow!("generating seed phrase failed"))?;
+                    let seed_words: String = seed_words.phrase().into();
+                    let sw_file = cmd::get_seed_words_file(wallet_dir);
+                    fs::write(sw_file.clone(), seed_words.clone())?;
+                    eprintln!("Wrote seeds words to {}", sw_file.display());
+                    WalletKey::SeedWordsFile {}
+                }
+            };
+            Config {
+                wallet_key,
+                ..Config::default_config(common_args.network)
+            }
+        }
+        InitOpt::Descriptor {
+            common_args,
+            psbt_output_dir,
+            external,
+            internal,
+        } => {
+            let psbt_dir = create_psbt_dir(wallet_dir, psbt_output_dir)?;
+            Config {
+                wallet_key: WalletKey::Descriptor { external, internal },
+                psbt_output_dir: psbt_dir,
+                ..Config::default_config(common_args.network)
+            }
+        }
+        InitOpt::XPub {
+            common_args,
+            psbt_output_dir,
+            xpub,
+        } => {
+            let psbt_dir = create_psbt_dir(wallet_dir, psbt_output_dir)?;
+
+            let mut external = String::from("wpkh(");
+            external.push_str(&xpub);
+            let mut internal = external.clone();
+            external.push_str("/0/*)");
+            internal.push_str("/1/*)");
 
-    let sw_file = cmd::get_seed_words_file(wallet_dir);
+            Config {
+                wallet_key: WalletKey::Descriptor {
+                    external,
+                    internal: Some(internal),
+                },
+                psbt_output_dir: psbt_dir,
+                ..Config::default_config(common_args.network)
+            }
+        }
+    };
+    fs::write(
+        config_file,
+        serde_json::to_string_pretty(&config.into_versioned())
+            .unwrap()
+            .as_bytes(),
+    )?;
 
-    fs::write(sw_file.clone(), seed_words.clone())?;
+    let mut random_bytes = [0u8; 64];
+    use rand::RngCore;
+    rand::rngs::OsRng.fill_bytes(&mut random_bytes);
 
-    eprintln!("Wrote seeds words to {}", sw_file.display());
-    println!("==== BIP39 seed words ====");
+    let hex_randomness = hex::encode(&random_bytes);
+    let mut secret_file = wallet_dir.to_path_buf();
+    secret_file.push("secret_protocol_randomness");
+    fs::write(secret_file, hex_randomness)?;
```

Comment:
:x:  I think we should not write this file unless we are using it. So only in the xpub/descriptor branch.

I just happened to stumble across this: https://github.com/bitcoin/bips/blob/master/bip-0085.mediawiki
Coldcard supports this and can export 64 bytes of hex: https://coldcard.com/docs/bip85

So when initing with coldcard we could also paste the randomness from there to here. I don't think it should be mandatory though randomly generating "secret_protocol_randomness" will probably be needed regardless.